base/file.jl

```diff
@@ -513,20 +513,45 @@ function mktemp(parent::AbstractString=tempdir(); cleanup::Bool=true)
     return (filename, Base.open(filename, "r+"))
 end
 
+# GUID  win32 api (consider # TODO: refactor)
+struct GUID
+    l1::Culong
+    w1::Cushort
+    w2::Cushort
+    b::NTuple{8,Cuchar}
+end
+
+# generates a random temporary string based on a UUID
+function _uiud_string()
+    id = Ref{GUID}()
+    r = ccall((:UuidCreate,:Rpcrt4), stdcall, Cint, (Ref{GUID},), id)
+    # ignore expected errors that we don't care about (these shouldn't even be an issue after Vista)
+    RPC_S_UUID_LOCAL_ONLY = 1824
+    RPC_S_UUID_NO_ADDRESS = 1739
+    if !(r == 0 || r == RPC_S_UUID_LOCAL_ONLY || r == RPC_S_UUID_NO_ADDRESS)
+        windowserror("UuidCreate", r % UInt32) # Throw on unexpected errors. Note, a RPC_STATUS is just a WINERROR with a different signedness of the type.
+    end
+
+    nameptr = Ref{Ptr{Cwchar_t}}()
+    r = ccall((:UuidToStringW, :Rpcrt4), stdcall, Cint, (Ref{GUID}, Ref{Ptr{Cwchar_t}}), id, nameptr)
+    r == 0 || windowserror("UuidToString", r % UInt32) # a RPC_STATUS is just a WINERROR with a different signedness of the type
+
+    namebuf = unsafe_wrap(Vector{Cwchar_t}, nameptr[], ccall(:wcslen, UInt, (Ptr{Cwchar_t},), nameptr[]))
+    name = transcode(String, namebuf)
+
+    ccall((:RpcStringFreeW, :Rpcrt4), stdcall, Cint, (Ref{Ptr{Cwchar_t}},), nameptr)
+
+    name = replace(name, '-' => '_')
```

Comment:
```suggestion
    n = UInt64(0)
    for c in name
        d = '0' <= c <= '9' ? c - '0' :
             'a' <= c <= 'f' ? c - 'a' :
             'A' <= c <= 'F' ? c - 'A' : continue
        n = (n << 4) + (d % UInt64)
    end
    slug = StringVector(10)
    chars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
    for i = 1:length(slug)
        slug[i] = chars[(n % length(chars)) + 1]
        n ÷= length(chars)
    end
    name = String(slug)
```

Reply:
a-z and A-Z generally map to the same files. should we just use the uppercase letters?

Reply:
Seems rather convoluted though to convert a UInt128 -> GUID -> String-base16 -> UInt64 -> String-base62. I would think it'd be much shorter to do:
```
id = Ref{UInt128}()
# UuidCreate(id)
name = string(id[] % UInt64, base=36, pad=5) # between 5 and 13 characters, to avoid special file names like COM1
name = name[1:min(end, 10)] # truncate to max of 10 characters (why 10? I have no idea, it came from the random number generator)
```

Reply:
Seems better to generate both upper and lowercase so the chance of collision is lower on systems with case sensitivity.  Of course, that increases the chance of collision on systems without it. Perhaps sample from a-z, A-Z with half the frequency of 0-9?

Reply:
@vtjnash  I like the simplicity of your proposal. What does `pad=5` do? 
I can update the PR to this change. It makes the code a lot shorter and simpler.

Reply:
It guarantees that it can't generate an invalid filename like `COM1`